src/TypeVisitor.ts

```diff
@@ -263,27 +263,38 @@ export default class TypeVisitor {
   }
 
   public getVariableType(variableName: string, node: t.Node): Type {
-    let mapping = this.symbolTable.getMap();
-    let type = mapping.get(variableName);
-    if (type != null) {
-      return type;
-    } else if (global.hasOwnProperty(variableName)) {
-      // This is a JS global (e.g. "console")
-      return new AnyType();
-    } else {
+    try {
+      if (global.hasOwnProperty(variableName)) {
+        // This is a JS global (e.g. "console")
+        return new AnyType();
+      }
+      return this.symbolTable.getVariableType(variableName);
+    } catch (e) {
```

Comment:
What exceptions is this try/catch meant to avoid?

Reply:
I throw an error in getVariableType if the variable isn't declared. This way I don't need to pass in node info and the error report into the symbolTable